src/main/java/com/team01/carsharingapp/service/impl/TelegramNotificationService.java

```diff
@@ -1,17 +1,78 @@
 package com.team01.carsharingapp.service.impl;
 
+import com.team01.carsharingapp.dto.rental.CreateRentalEvent;
+import com.team01.carsharingapp.model.Rental;
+import com.team01.carsharingapp.repository.RentalRepository;
 import com.team01.carsharingapp.service.NotificationService;
 import com.team01.carsharingapp.telegramapi.Bot;
+import java.time.LocalDate;
+import java.util.Comparator;
+import java.util.List;
+import java.util.function.Function;
+import java.util.stream.Collectors;
 import lombok.RequiredArgsConstructor;
+import org.springframework.context.event.EventListener;
+import org.springframework.scheduling.annotation.EnableScheduling;
+import org.springframework.scheduling.annotation.Scheduled;
 import org.springframework.stereotype.Service;
 
 @Service
 @RequiredArgsConstructor
+@EnableScheduling
 public class TelegramNotificationService implements NotificationService {
+    private static final String NO_OVERDUE_MESSAGE = "No rentals overdue today!";
+    private static final String OVERDUE_MESSAGE =
+            "Customer - %s, expected return date - %s, car - %s";
+    private static final String CREATE_RENTAL_MESSAGE =
+            "NEW RENTAL : Customer - %s, expected return date - %s, car - %s";
+    private static final String CRON = "0 0 12 * * ?";
     private final Bot bot;
+    private final RentalRepository rentalRepository;
 
     @Override
     public void sendNotification(String string) {
         bot.sendMessage(string);
     }
+
+    @Scheduled(cron = CRON)
+    private void sendDailyStatistic() {
+        LocalDate date = LocalDate.now().plusDays(1);
```

Comment:
Magic number 🪄

Reply:
And do you need that variable separately?